### ISSUE TITLE
The purpose of defaultView

### ISSUE THREAD

**Issue by tsirolnik** [tmpvar/jsdom#1685]
Hi, the purpose of defaultView is not clear. 

The MDN states the following - 
`In browsers, document.defaultView returns the window object associated with a document, or null if none is available.`[1]

But the thing that actually happens in jsdom is that the jsdom.jsdom function returns *window.document* and, in order to use the window(?) as said in the README.md we need to use window.document.defaultView which, would normally refrence to the first window. But, again, things are weird and window != window.document.defaultView.

So

```js
window.fooa = 'bar';
window.document.defaultView.foob = 'bar';
window.document.fooa = 'bar';
window.document.defaultView.document.foob = 'bar';

window.fooa === window.document.defaultView.foob
window.document.fooa === window.document.defaultView.document.foob
```
would return false, though it should return true.

So, basically, defaultView, from what I'm experiencing, isn't circular.

Another issue is the executed JS is using the Window object but the inner workings of jsdom (Loading resources and etc) use other object (defaultView or something else). This was encountered when using a new Location object as window.location and window._URL, but the internals use the window.defaultView._URL

To make the users and contributors of jsdom more comfortable with this great library, I think that additionally, a UML like of a jsdom instance and a flow of initialization (jsdom.jsdom -> new Window, etc) would be nice to have.

[1] - https://developer.mozilla.org/en-US/docs/Web/API/Document/defaultView

**Comment by domenic** [comment]
```js
"use strict";
const jsdom = require("jsdom");

const document = jsdom.jsdom();
const window = document.defaultView;

window.fooa = 'bar';
window.document.defaultView.foob = 'bar';
window.document.fooa = 'bar';
window.document.defaultView.document.foob = 'bar';

console.log(window.fooa === window.document.defaultView.foob); // logs true
console.log(window.document.fooa === window.document.defaultView.document.foob); // logs true
```

Happy to reopen if you can provide a repro like the above somehow.
